tools/plugins/mgm/src/main/kotlin/net/corda/cli/plugins/mgm/BaseOnboard.kt

```diff
@@ -335,15 +350,19 @@ abstract class BaseOnboard : Runnable {
     private fun waitForFinalStatus(id: String) {
         val end = System.currentTimeMillis() + 5 * 60 * 1000
         while (System.currentTimeMillis() < end) {
+            Thread.sleep(400)
             val status = Unirest.get("/membership/$holdingId/$id").asJson()
             val registrationStatus = status.bodyOrThrow().`object`.get("registrationStatus")
-            if (registrationStatus == "APPROVED") {
-                return
-            } else if (registrationStatus =="DECLINED") {
-                throw OnboardException("Registration has been declined.")
-            } else {
-                println("Status of $x500Name registration is $registrationStatus")
-                Thread.sleep(400)
+            when (registrationStatus) {
+                "APPROVED" -> {
+                    return
+                }
+                "DECLINED", "INVALID" -> {
+                    throw OnboardException("Registration has been $registrationStatus.")
```

Comment:
```suggestion
                    throw OnboardException("Status of registration is $registrationStatus.")
```
otherwise we'd end up with "Registration has been INVALID"